src/System.Net.Security/src/System/Net/SslStreamPal.Windows.cs

```diff
@@ -126,8 +127,13 @@ public static SafeFreeCredentials AcquireCredentialsHandle(X509Certificate certi
             return AcquireCredentialsHandle(direction, secureCredential);
         }
 
-        public static SecurityStatusPal EncryptMessage(SafeDeleteContext securityContext, byte[] writeBuffer, int size, int headerSize, int trailerSize, out int resultSize)
+        public static SecurityStatusPal EncryptMessage(SafeDeleteContext securityContext, byte[] input, int offset, int size, int headerSize, int trailerSize, ref byte[] output, out int resultSize)
         {
+            byte[] writeBuffer = output;
+
+            // Copy the input into the output buffer to prepare for SCHANNEL's expectations
+            Buffer.BlockCopy(input, offset, writeBuffer, headerSize, size);
+
```

Comment:
Is there a test we can add that would have failed before this fix and succeeds after?